src/ekat/util/ekat_rational_constant.hpp

```diff
@@ -36,8 +32,12 @@ constexpr long long gcd (const long long a, const long long b) {
 //       would probably require too much effort, especially
 //       considered the scope of the usage of this class
 
-class RationalConstant {
-public:
+struct RationalConstant {
+
+  using iType = long long;
+
+  const iType num;
+  const iType den;
```

Comment:
What do you think about `denom` instead of `den`? I'm just thinking about all the technical language that might be embedded in a given expression involving a rational constant, but maybe I'm overthinking it.

Reply:
I liked having the same short length for both (I would even have gone for `n` and `d`), but have no strong feeling. Do you think there might be ambiguities with the name 'den'?

Reply:
Probably not. There is always "density", which appears everywhere, but likely not in this context. I guess it's more of a cadence between the syllables in **num**erator and **denom**inator, but this is very subjective. I don't have strong feelings. :-)

Reply:
Yeah, I don't think this class will be used outside the context of the Units framework impl, so ppl should not even see these when using Units in their code...